src/cli/utils/getRoute.ts

```diff
@@ -12,28 +21,43 @@ function removeGroupSegments(input: string) {
 }
 
 function removeParallelSegments(input: string) {
-  return input.replace(/\/@[\w]+/g, '')
+  return input.replace(/\/@\w+/g, '')
+}
+
+function removeInterceptedSegments(input: string) {
+  let result = input.replace(/\(\.\)/g, '')
+  const twoDotsRegExp = /[[\w-\]]+\/\(\.{2}\)/g
+  while (twoDotsRegExp.test(result)) {
+    result = result.replace(twoDotsRegExp, '')
+  }
+  return result.replace(/.*\(\.{3}\)/g, '/')
 }
 
 function formatDynamicSegments(input: string) {
   return input.replace(/\/\[(\w+)\]/g, '/:$1')
 }
 
 function getRouteName({ originPath }: Rewrite) {
-  const formatRouteName = pipe(removePageSegment)
+  const formatRouteName = pipe(
+    removePageSegment,
+    removeGroupSegments,
+    removeParallelSegments,
+    removeInterceptedSegments
+  )
   return asRootPath(formatRouteName(originPath))
 }
 
 function getRouteHref({ localizedPath }: Rewrite) {
+  const localeSegment = extractLocaleSegment(localizedPath)
   const formatRouteHref = pipe(
+    removeLocaleSegment,
     removePageSegment,
     removeGroupSegments,
     removeParallelSegments,
-    formatDynamicSegments,
-    asRootPath
+    removeInterceptedSegments,
+    formatDynamicSegments
   )
-
-  return formatRouteHref(localizedPath)
+  return asRootPath(localeSegment + formatRouteHref(localizedPath))
```

Comment:
```suggestion
  return asRootPath(localeSegment, formatRouteHref(localizedPath))
```

IMO this is a bit more safe.

Reply:
Fixed 👍